src/solvers/solver_utils.hpp

```diff
@@ -0,0 +1,126 @@
+//========================================================================================
+// (C) (or copyright) 2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef SOLVERS_SOLVER_UTILS_HPP_
+#define SOLVERS_SOLVER_UTILS_HPP_
+
+#include <string>
+#include <vector>
+
+#include "kokkos_abstraction.hpp"
+
+namespace parthenon {
+
+namespace solvers {
+
+struct SparseMatrixAccessor {
```

Comment:
A little documentation on what this struct is useful for would be good (I assume it's not yet documented because it's WIP)

Reply:
yeah, I'll document.  just haven't gotten there yet.  in brief, this class let's us define sparse matrices via all our variable/packing machinery.

Reply:
I updated docs

Reply:
I think it would be better that the classes and code themselves are documented with comments explaining how they function and operate, but looking at the rest of the code that's not the norm. So this is ok